gg_fit_curve.py

```diff
@@ -0,0 +1,402 @@
+"""
+Python transcription of https://github.com/erich666/GraphicsGems/blob/master/gems/FitCurves.c
+as it existed 2/24/2020.
+"""
+from math import sqrt
+from typing import NamedTuple, Sequence, Tuple
+
+
+class Point(NamedTuple):
+    x: float
+    y: float
+
+    def magnitude(self) -> float:
+        return sqrt(self.x * self.x + self.y * self.y)
+
+    # V2AddII; it's weird this works on vectors...
+    def add(self, other) -> "Point":
+        return Point(self.x + other.x, self.y + other.y)
+
+    # V2SubII
+    def sub(self, other) -> "Point":
+        return Point(self.x - other.x, self.y - other.y)
+
+    def scale(self, s: float) -> "Point":
+        return Point(self.x * s, self.y * s)
+
+    @staticmethod
+    def tan_left(points: Sequence["Point"]) -> "Vector":
```

Comment:
actually it's better to have separate functions instead of methods of either Point or Vector. These three are very functions to compute left, right and center tangent vectors for our bezier, not generic constructors for a Vector.

Reply:
A style related remark, highly subjective, so feel free to ignore it. I notice you tend to give noun-like names to functions/methods that return something, in this case `tan_left`. In general I prefer to have some verb in the name of a function or method, in this case `compute_left_tangent` (as in the original c code). You may argue it's redundant. But sometimes it's nice to be able to distinguish a function from the local variable which is assigned its return value. E.g. `tan_left = compute_left_tangent(points)`.

Reply:
I will admit to disliking generic prefixes. Having 101 methods with compute_ in front of them doesn't help much imo. Another idea would be to put them into a static method class, resulting in something like Tan.left(...)

Reply:
Fixed during conversion to picosvg geometric types